components/support/jwcrypto/src/lib.rs

```diff
@@ -2,42 +2,67 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-//! Theorically, everything done in this crate could and should be done in a JWT library.
-//! However, none of the existing rust JWT libraries can handle ECDH-ES encryption, and API choices
-//! made by their authors make it difficult to add this feature.
-//! In the past, we chose cjose to do that job, but it added three C dependencies to build and link
-//! against: jansson, openssl and cjose itself.
+//! A library for using Javascript Web Tokens (JWK) and related concepts.
+//! The encryption is done by [rc_crypto] - this crate just does the
+//! Javascript/JSON etc wrappers around this crypto. As a result, most of the
+//! structs etc here support serialization and deserialization to and from JSON
+//! via serde.
+
+// Theoretically, everything done in this crate could and should be done in a JWT library.
+// However, none of the existing rust JWT libraries can handle ECDH-ES encryption, and API choices
+// made by their authors make it difficult to add this feature.
+// In the past, we chose cjose to do that job, but it added three C dependencies to build and link
+// against: jansson, openssl and cjose itself.
+// So now, this *is* our JWT library.
 
 pub use error::JwCryptoError;
 use error::Result;
 use rc_crypto::agreement::EphemeralKeyPair;
 use serde_derive::{Deserialize, Serialize};
 use std::str::FromStr;
 
+mod aes;
+mod direct;
 pub mod ec;
 mod error;
 
+/// Specifies the mode, algorithm and keys of the encryption operation.
 pub enum EncryptionParameters<'a> {
     // ECDH-ES in Direct Key Agreement mode.
     #[allow(non_camel_case_types)]
     ECDH_ES {
         enc: EncryptionAlgorithm,
         peer_jwk: &'a Jwk,
     },
+    // Direct Encryption with a shared symmetric key.
+    Direct {
+        enc: EncryptionAlgorithm,
```

Comment:
I ended up adding support for the algo here, just for symmetry with the EC params, but happy to remove it if you think it just causes noise. In particular, the constructors for the `Direct` keys do not support this being variable, which is an argument for *something* changing here, but 🤷

Reply:
The ECDH constructors shouldn't let it be variable either because we don't support any other value, but I'm fine with keeping it here, I think it conceptually makes sense.